### PR TITLE
Add webonyx/graphql-php version 0.13 compatibility

### PR DESCRIPTION
**Pull request by antograssiot**
~~waiting for https://github.com/symfony/symfony/pull/29350~~

| Q             | A
| ------------- | ---
| Bug fix?      | kind of
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Allow the test suite to run normaly.

### CODE REVIEW

**Comment by antograssiot** [comment]
I'll have a look to the other PHPStan issue later if we are going to support several version of `webonyx/graphql-php` as I started.

**Comment by alanpoulain** [comment]
Wouldn't it be better to force `webonyx/graphql-php` to be >=0.13?

**Comment by alanpoulain** [comment]
The previous update was done by forcing the next version.
It's because the GraphQL part of API Platform is an experimental feature. It allows us to do some breaking changes.
In this case, I am not sure about what to do. The code will be less readable with the `method_exists` but it's mainly in the tests.
WDYT @api-platform/core-team?

**Comment by bendavies** [comment]
ah that's the issue i raised:
https://github.com/symfony/symfony/issues/29340

what problem is that causing in api-platform?

**Comment by dunglas** [comment]
Regarding GraphQL, I'm all for bumping the version. It is still marked as `@experimental` anyway, and it'll stay like that until webonyx/graphql releases a 1.0 version.

**Comment by antograssiot** [comment]
ok I'll update tomorrow

**Comment by antograssiot** [comment]
@bendavies yes that's the issue you raisd. I broke [this test](https://travis-ci.org/api-platform/core/jobs/459908728#L689) (`children` was an empty array) it also broke nelmio fixture loading in some testsuite for example (that's how I discovered it in the first place)

@alanpoulain I've bump `webonyx/graphql-php` as discussed

File: src/GraphQl/Type/SchemaBuilder.php [hunk 1]
```diff
@@ -204,7 +204,7 @@ private function getResourceFieldConfiguration(string $resourceClass, ResourceMe
             }
 
             $graphqlWrappedType = $graphqlType instanceof WrappingType ? $graphqlType->getWrappedType() : $graphqlType;
-            $isInternalGraphqlType = \in_array($graphqlWrappedType, GraphQLType::getInternalTypes(), true);
+            $isInternalGraphqlType = \in_array($graphqlWrappedType, GraphQLType::getStandardTypes(), true);
```
Comment (alanpoulain) [inline src/GraphQl/Type/SchemaBuilder.php]: Shouldn't we rename the variables as well?

File: phpstan.neon [hunk 2]
```diff
@@ -44,3 +44,7 @@ parameters:
         - '#Method Symfony\\Component\\Serializer\\Normalizer\\AbstractObjectNormalizer::__construct\(\) invoked with 6 parameters, 0-4 required\.#'
         - '#Parameter \#1 \$bitmask of class Symfony\\Component\\Serializer\\Encoder\\JsonEncode constructor expects int, array<string, int> given\.#'
         - '#Parameter \#1 \$associative of class Symfony\\Component\\Serializer\\Encoder\\JsonDecode constructor expects bool, array<string, true> given\.#'
+
+        # Expected, due to fixture detection in webonyx/graphql-php
+        - '#Parameter \#2 \$errors of class GraphQL\\Executor\\ExecutionResult constructor expects array<GraphQL\\Error\\Error>, array<int\|string, Exception\|string> given\.#'
```
Comment (alanpoulain) [inline phpstan.neon]: I'm not sure if this line:
https://github.com/api-platform/core/blob/master/src/GraphQl/Action/EntrypointAction.php#L67
is still working.
I think it should be interesting to add a test.

Comment (antograssiot) [reply phpstan.neon]: indeed, removing it doesn't break any existing tests IIRC so I wasn't sure.
I'll dig into that

File: phpstan.neon [hunk 3]
```diff
@@ -44,3 +44,7 @@ parameters:
         - '#Method Symfony\\Component\\Serializer\\Normalizer\\AbstractObjectNormalizer::__construct\(\) invoked with 6 parameters, 0-4 required\.#'
         - '#Parameter \#1 \$bitmask of class Symfony\\Component\\Serializer\\Encoder\\JsonEncode constructor expects int, array<string, int> given\.#'
         - '#Parameter \#1 \$associative of class Symfony\\Component\\Serializer\\Encoder\\JsonDecode constructor expects bool, array<string, true> given\.#'
+
+        # Expected, due to fixture detection in webonyx/graphql-php
```
Comment (alanpoulain) [inline phpstan.neon]: fixture detection?

**Comment by antograssiot** [comment]
yeah, I just ran out of time this morning @alanpoulain

File: composer.json [hunk 4]
```diff
@@ -20,7 +20,7 @@
         "psr/container": "^1.0",
         "symfony/http-foundation": "^3.1 || ^4.0",
         "symfony/http-kernel": "^2.7 || ^3.0 || ^4.0",
-        "symfony/property-access": "^2.7 || ^3.0 || ^4.0",
+        "symfony/property-access": "^2.7 || ^3.0 <3.4.19 || ^4.0 <4.1.8",
```
Comment (Taluu) [inline composer.json]: as there has been a fix, shouldn't these be declared as conflict ?

Comment (antograssiot) [reply composer.json]: I'll just remove this a 4.2 is here anyway

**Comment by antograssiot** [comment]
Tests added @alanpoulain 
I've also removed the version constraint update for Symfony and update the PHPStan rule to follow the Symfony 4.2 release
It should be good this time

i did cleanup some old `@dropSchema` annotation that were still there at the same time

File: composer.json (hunk 4 above)
Comment (Taluu) [reply composer.json]: Yeah but these 2 versions still conflicts somehow. Or you can categorize them as "yup doesn't work we know" ?

**Comment by alanpoulain** [comment]
Thank you @antograssiot :slightly_smiling_face:

**Pull request merged**